src/network/getters/getSponsorshipStats.ts

```diff
@@ -0,0 +1,110 @@
+import moment from 'moment'
+import { ChartPeriod } from '~/shared/components/NetworkChart/NetworkChart'
+import { GetSponsorshipDailyBucketsQuery } from '~/generated/gql/network'
+import { getSponsorshipDailyBuckets } from '~/network/getters/getSponsorshipDailyBuckets'
+import { toBN } from '~/utils/bn'
+import { getSponsorshipTokenInfo } from '~/network/getters/getSponsorshipTokenInfo'
+import { TokenInformation } from '~/marketplace/utils/web3'
+
+export const getSponsorshipStats = async (
+    sponsorshipId: string,
+    selectedPeriod: ChartPeriod,
+    dataSource: string,
+    ignoreToday?: boolean,
+): Promise<{ x: number; y: number }[]> => {
+    const tokenInfo = (await getSponsorshipTokenInfo()) as TokenInformation
+    const todayUTC = moment().utc().startOf('day').format('X')
+    let result: GetSponsorshipDailyBucketsQuery['sponsorshipDailyBuckets']
+    switch (selectedPeriod) {
+        case ChartPeriod.SevenDays:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                7,
+                0,
+                ignoreToday ? todayUTC : undefined,
```

Comment:
Repetition.

Maybe have a const that's a `number | undefined` and evaluate `ignoreToday` once?

Reply:
good idea